spec/commands/install_spec.rb

```diff
@@ -506,4 +506,24 @@
       expect(out).not_to include("You're running Bundler #{Bundler::VERSION} but this project uses #{Bundler::VERSION}.")
     end
   end
+
+  describe "when sharing gemfile with path sources across different directories" do
+    it do
+      build_lib("foo", :path => "foo")
+      create_file("app/Gemfile")
+      create_file("app/.bundle/config", <<-YAML)
+        ---
+        BUNDLE_GEMFILE: "../Gemfile"
+      YAML
+      gemfile <<-RUBY
+        gem 'foo', path: 'foo'
+      RUBY
+
+      bundle "install"
+      in_app_root_custom "app" do
+        bundle "install"
+        expect(out).to include("Bundle complete!")
```

Comment:
`Bundler.root` should probably be changed instead of creating this empty `Gemfile`: it should choose the first directory containing `Gemfile`, `gems.rb` or `.bundle`.